torch/csrc/jit/script/compiler.cpp

```diff
@@ -2482,6 +2496,29 @@ TypePtr parseTypeFromExpr(Expr expr) {
                                   << " cannot be used in a type expression";
 }
 
+c10::optional<IValue> parseValueFromExpr(Expr tree) {
+  switch (tree.kind()) {
+    case TK_CONST: {
+      Const c(tree);
+      if (c.isFloatingPoint()) {
+        return c.asFloatingPoint();
+      } else {
+        return c.asIntegral();
+      }
+    } break;
+    case TK_TRUE: {
+      return true;
+    } break;
+    case TK_FALSE: {
+      return false;
+    } break;
+    case TK_NONE: {
+      return IValue();
```

Comment:
These rules are very easy to get out of sync with the compiler rules for the same constants. They are not always correct either. For instance None handling is wrong here when None is used for n Option[Tensor] or generator. https://github.com/pytorch/pytorch/pull/14176  fully implements default values using the compiler and we should its implementation instead.